blog/2022-05-04-announcing-stryker-6-effortless-sauce-mutation.md

```diff
@@ -0,0 +1,257 @@
+1. Report the result
+
+As you can see, _all files are loaded and unloaded_ for each mutant run. This way of reloading [won't ever work with native ESM](https://github.com/nodejs/tooling/issues/51), as there is simply no `cache` we can `delete` the loaded files from 🤷‍♀️.
+
+The reloading is also expensive. Think about it: if your project contains 1k files, with a total of 10k mutants, it could mean a total of _ten million file IO actions_. That's why mutation testing with StrykerJS would slow down considerably when the size of your project grows.
+
+When you think about it, it is also **unnecessary**; why not simply keep all files loaded? Since they contain all the mutants anyway. This new process looks something like this:
+
+(new process)
+
+1. (once) Load the code using `import`
+1. Activate the mutant
+1. Run the tests with the test runner of choice.
+1. Report the result
+
+The result is a significant performance improvement 🐇. You can clearly see the difference when running StrykerJS on Stryker's core itself:
+
+| Old                                         |
+| ------------------------------------------- |
+| ![stryker-v5](/images/blogs/stryker-v5.gif) |
+
+| New \*                                      |
+| ------------------------------------------- |
+| ![stryker-v6](/images/blogs/stryker-v6.gif) |
+
+\* using `--ignoreStatic`
+
+That's a whopping 70% performance improvement 🤯.
+
+We call this feature "hot reload," and StrykerJS supports it in `@stryker-mutator/mocha-runner` and `@stryker-mutator/jasmine-runner`. Please note that this feature is always on from version 6 onward, even for plain old CommonJS style projects! It does require a higher version of mocha or jasmine, respectively, see [breaking changes](#-breaking-changes). Support for `@stryker-mutator/cucumber-runner` for the release of version 8 of `@cucumber/cucumber`.
+
+Unfortunately, we're not supporting hot reload for the `@stryker-mutator/jest-runner` or `@stryker-mutator/karma-runner` yet; please up-vote [#3455](https://github.com/stryker-mutator/stryker-js/issues/3455) or [#3454](https://github.com/stryker-mutator/stryker-js/issues/3454) respectively if you want to see it happen 💙.
+
+## 🗿 Static mutant improvements
+
+Yes, indeed, hot reload sounds incredible! But what about _static mutants_?
+
+A static mutant is a mutant that is executed once on _file load_ rather than during _test run-time_.
+
+Take this small example:
+
+```js
+// greet.js
+const hi = '👋';
+
+export function greet(name) {
+  return `${hi} ${name}`;
+}
+
+// greet.spec.js
+import { greet } from './greet.js';
+
+it('should greet me', () => {
+  expect(greet('me')).toBe('👋 me');
+});
+```
+
+When you run StrykerJS on this code, it will create a mutant for the `hi` constant:
+
+```diff
+-const hi = '👋';
++const hi = '';
+```
+
+When you consider mutant schemata, the actual code produced by StrykerJS looks more like this:
+
+```js
+const hi = global.activeMutant === '1' ? '' : '👋';
```

Comment:
Can't you solve this issue by replacing const with let? That changes initialisation to runtime right? You could use the hot reload feature for all mutants then because you don't have any static mutants left. Or am I missing something 🙈

Reply:
You are missing something :)

No it wouldn't. The code is executed at file load, rather than test runtime. Comparable to a `static {}` block in C#. Should I clarify it?